offchain-modules/packages/scripts/src/utils/cardano_batch_test.ts

```diff
@@ -0,0 +1,125 @@
+import * as CardanoWasm from '@emurgo/cardano-serialization-lib-nodejs';
+import { asyncSleep } from '@force-bridge/x/dist/utils';
+import { logger } from '@force-bridge/x/dist/utils/logger';
+import CKB from '@nervosnetwork/ckb-sdk-core';
+import { WalletServer, Seed, AddressWallet, ShelleyWallet } from 'cardano-wallet-js';
+import { JSONRPCClient } from 'json-rpc-2.0';
+import fetch from 'node-fetch/index';
+import { prepareCkbPrivateKeys, prepareCkbAddresses, burn } from './eth_batch_test';
+
+export async function cardanoBatchTest(
+  ckbPrivateKey: string,
+  WALLET_SERVER_URL: string,
+  ckbNodeUrl: string,
+  ckbIndexerUrl: string,
+  forceBridgeUrl: string,
+  adaForceBridgeAddr: string,
+  adaWalletMnemonic: string,
+  batchNum = 100,
+  lockAmount = 2000000,
+  burnAmount = '1990000',
+): Promise<void> {
+  logger.info('adaBatchTest start!');
+  const ckb = new CKB(ckbNodeUrl);
+
+  const client = new JSONRPCClient((jsonRPCRequest) =>
+    fetch(forceBridgeUrl, {
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json',
+      },
+      body: JSON.stringify(jsonRPCRequest),
+      // id: 1,
+    }).then((response) => {
+      if (response.status === 200) {
+        // Use client.receive when you received a JSON-RPC response.
+        return response.json().then((jsonRPCResponse) => client.receive(jsonRPCResponse));
+      } else if (jsonRPCRequest.id !== undefined) {
+        return Promise.reject(new Error(response.statusText));
+      }
+    }),
+  );
+
+  logger.info('adaBatchTest adaWallet created');
+
+  const ckbPrivs = await prepareCkbPrivateKeys(batchNum);
+  const ckbAddresses = await prepareCkbAddresses(ckb, ckbPrivs, ckbPrivateKey, ckbNodeUrl, ckbIndexerUrl);
+
+  const passphrase = 'user_wallet_passphrase';
+  const adaWallet = await getUserWallet(WALLET_SERVER_URL, adaWalletMnemonic, passphrase);
+
+  const lockTxs = await lock(client, adaWallet, passphrase, ckbAddresses, adaForceBridgeAddr, lockAmount, 10 * 1000);
+  // await check(client, lockTxs, ckbAddresses, batchNum, adaTokenAddress);
+
+  await asyncSleep(140000);
+  const adaAddress =
+    'addr_test1qzr32f6d58c3fud3u8zsqj6gnnmu06f3rkgt9z6qpa05prc4a05fk35vqa77wdtvrllelfa3rk0tn8g9kgvhks8983nsrhgv65';
+
+  {
+    const maxTryTimes = 5;
+    for (let tryTime = 0; ; tryTime++) {
+      try {
+        const burnTxs = await burn(ckb, client, ckbPrivs, ckbAddresses, adaAddress, 'ada', burnAmount, 'Cardano', 0);
```

Comment:
We should also check the status of burnTxs either before we say `adaBatchTest pass!`

Reply:
Yes, this should be done, but the dev work is currently blocked due to the issue in running the integration test because of burn tx creation failure.